docs/utilities/idempotency.md

```diff
@@ -884,6 +960,49 @@ When creating the `DynamoDBPersistenceStore`, you can set a custom [`DynamoDbCli
         .build();
     ```
 
+### Customizing Redis client
+
+The `RedisPersistenceStore` uses the [`JedisPooled`](https://www.javadoc.io/doc/redis.clients/jedis/latest/redis/clients/jedis/JedisPooled.html) java client to connect to the Redis standalone server or the [`JedisCluster`](https://javadoc.io/doc/redis.clients/jedis/4.0.0/redis/clients/jedis/JedisCluster.html) to connect to the Redis cluster.
+When creating the `RedisPersistenceStore`, you can set a custom Jedis client:
+
+=== "Custom JedisPooled with connection timeout"
+
+    ```java hl_lines="2-11 13 18"
+    public App() {
+        JedisConfig jedisConfig = JedisConfig.Builder.builder()
+            .withHost(redisCluster.getHost())
+            .withPort(redisCluster.getBindPort())
+            .withJedisClientConfig(DefaultJedisClientConfig.builder()
+                        .user("user")
+                        .password("secret")
+                        .ssl(true)
+                        .database(1)
+                        .connectionTimeoutMillis(3000)
+            .build())
+        .build();
+
+        JedisPooled jedisPooled = new JedisPooled(new HostAndPort("host",6789), jedisConfig)
+
+        Idempotency.config().withPersistenceStore(
+          RedisPersistenceStore.builder()
+                .withKeyPrefixName("items-idempotency")
+                .withJedisClient(jedisPooled)
+                .build()
+      ).configure();
+    }
+    ```
+
+!!! info "Default configuration is the following:"
```

Comment:
What do you mean by "default configuration", it is what we use as default ? If that's the case, does it make sense? Not sure anyone is using the user "default" with no password...

I think we should ask for all the required fields to create a JedisClient ourself:
- host (mandatory)
- port (default: 6379)
- user (mandatory)
- password (mandatory)
- ssl (default: false)
- database ?
- something else ?

@scottgerring wdyt ?

Reply:
Default here means, as provided by the jedis client [ default config](https://javadoc.io/doc/redis.clients/jedis/3.7.0/redis/clients/jedis/DefaultJedisClientConfig.Builder.html). 
In Redis before version 6, you could [only require password for auth ](https://redis.io/commands/auth/). 
Also, by default all access is unathenticated and a user with name "default" is created. So, by default (that is the redis default), you can execute commands either without any user/pass or with user:"default" and an empty pass. The user "default" is created even after redis version 6, for backwards compatibility purposes.
See [ACL redis doc](https://redis.io/docs/management/security/acl/). 
For these reasons, I am not sure we should make them mandatory.

Actually I have to update the powertools documentation, since user/pass, by default, in JedisClient are null.

Reply:
Found also this extract here from [redis.conf ](https://raw.githubusercontent.com/redis/redis/7.2/redis.conf)

> # The special username "default" is used for new connections. If this user
> # has the "nopass" rule, then new connections will be immediately authenticated
> # as the "default" user without the need of any password provided via the
> # AUTH command. Otherwise if the "default" user is not flagged with "nopass"
> # the connections will start in not authenticated state, and will require
> # AUTH (or the HELLO command AUTH option) in order to be authenticated and
> # start to work.